### DISCUSSION TITLE
Cutting image in half with both saved sides

### DISCUSSION THREAD

**Discussion by fuchsf-maker** [ImageMagick/ImageMagick#6760] [Help]
### ImageMagick version

7.1.1-15

### Operating system, version and so on

Win10

### Description

Hello!

I am having a hard time cutting an image in half. I am new to using commandlines. 
I want to cut multiple images in half, so I have both sides saved in a correct order. I have tried out multiple coding, but I can't seem to get the right one. 

This is the one I am using atm: 
`FOR /R "InputPath" %f IN (*.tif) DO D:\ImageMagick-7.1.1-15-portable-Q16-x64\magick convert "%f" -crop 50%x100% +repage "OUTPUT\page_%d.tif"`

This results into cutting all, but saving only the left side + the right side of the very last image. So out of 4 test images I get 5 pictures. 

Second thing I need is that I the pages should be in the right order. Like this:
1st pic, left side
1st pic, right side
2nd pic, left side
2nd pic, right side 

3rd thing: What would be great is that I want imagemagick to find any .tif in a directionary (I have multiple papers that I need to cut) and put each one into a seperate folder if possible. 

Does anyone have a solution for that? I got lost within the details. :-(

Thank you!

**Comment by snibgo** [comment]
I suggest removing `convert` from your command. Using `magick convert` gives the old v6 syntax. Using just `magick` will use the much better v7 syntax.

You run one magick command per input file. That is fine, but the `%d` counter initializes to zero at each command. So I would expect that each input file makes two outputs: "OUTPUT\page_0.tif" and "OUTPUT\page_1.tif", for the left and right respectively. So each output pair has the same names as the previous pair, overwriting them.

I don't understand how "So out of 4 test images I get 5 pictures."

I suggest using different output filenames, based on the input filenames. I suggest "%~nf_%d.tif".

You can put each pair in a separate directory, but the directory must already exist. Within the `DO` loop, first do a `md` command to make the directory, then run the magick command.

What directory names do you want? You could use `%~nf` to also name the directory after the name of the input file (without drive, directory or extension).

```
FOR /R "InputPath" %f IN (*.tif) DO (
  md %~nf
  D:\ImageMagick-7.1.1-15-portable-Q16-x64\magick ^
  "%f" ^
  -crop 50%x100% +repage "%~nf\%~nf_%d.tif"
)
```
I haven't tested this, and may have made a mistake.

If you put this in a BAT script file, double each `%`.
